types/jest/jest-tests.ts

```diff
@@ -1498,6 +1498,32 @@ test.each([[1, 1, 2], [1, 2, 3], [2, 1, 3]])(
     5000
 );
 
+const constCases: [['a', 'b', 'ab'], ['d', 2, 'd2']] = [['a', 'b', 'ab'], ['d', 2, 'd2']];
+test.each(constCases)('%s + %s', (s1, s2, expected) => {
+    expect(s1 + s2).toBe(expected);
+});
+
+const constCasesWithMoreThanTen: [
+  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11],
+  [91, 92, 93, 94, 95, 96, 97, 98, 99, 910, 911]
+] = [
+  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11],
+  [91, 92, 93, 94, 95, 96, 97, 98, 99, 910, 911]
+];
+test.each(constCasesWithMoreThanTen)('should fall back with more than 10 args', (n1, n2, n3, n4, n5, n6, n7, n8, n9, n10, n11) => {
```

Comment:
This&nbsp;should test&nbsp;that the&nbsp;`args`&nbsp;array is&nbsp;still&nbsp;correctly a&nbsp;tuple, even&nbsp;for&nbsp;overly long&nbsp;tuples.

```suggestion
test.each(constCasesWithMoreThanTen)('should fall back with more than 10 args', (...args) => {
    args; // $ExpectType [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11] | [91, 92, 93, 94, 95, 96, 97, 98, 99, 910, 911]
    const [n1, n2, n3, n4, n5, n6, n7, n8, n9, n10, n11] = args;
```

Reply:
tried putting in this but seems like union types are really flaky.
https://github.com/microsoft/dtslint/issues/191